type.go

```diff
@@ -302,6 +323,39 @@ func NewEquipmentDetail(es EquipementSource, updatedAt time.Time, location *time
 	}, nil
 }
 
+//FreeFloating defines how the object is represented in a response
+type Coord struct {
+	Lat float64 `json:"lat,omitempty"`
+	Lon float64 `json:"lon,omitempty"`
+}
+
+type FreeFloating struct {
+	PublicId string `json:"public_id,omitempty"`
+	ProviderName string `json:"provider_name,omitempty"`
+	Id string `json:"id,omitempty"`
+	Type string `json:"type,omitempty"`
+	Coord Coord `json:"coord,omitempty"`
+	Propulsion string `json:"propulsion,omitempty"`
+	Battery int `json:"battery,omitempty"`
+	Deeplink string `json:"deeplink,omitempty"`
+	Attributes [] string `json:"attributes,omitempty"`
+}
+
+// NewFreeFloating creates a new FreeFloating object from the object Vehicle
+func NewFreeFloating(ve Vehicle) (*FreeFloating, error) {
+	return &FreeFloating{
+		PublicId:		ve.PublicId,
+		ProviderName: 	ve.Provider.Name,
+		Id:				ve.Id,
+		Type:			ve.Type,
+		Coord:			Coord{Lat: ve.Latitude, Lon: ve.Longitude},
+		Propulsion:		ve.Propulsion,
+		Battery:		ve.Battery,
+		Deeplink:		ve.Deeplink,
+		Attributes:		ve.Attributes,
+	}, nil
```

Comment:
You don't need to return `nil` here, as it doen't mean anything.
```suggestion
	}
```